### ISSUE TITLE
Support client configuration via environmental variables

### ISSUE THREAD

**Issue by gyuho** [etcd-io/etcd#12785]
As of etcd 3.4, we support `ETCD_CLIENT_DEBUG` to configure logging level.

https://github.com/etcd-io/etcd/blob/2702f9e5f21193d8faffe3b6ae57dfc2dab662bd/clientv3/client.go#L50-L55

We should make this more generic to support other options.

One use case is to configure our auto-sync interval, as it becomes harder to expose individual configuration via flags (e.g., `kube-apiserver` needs to add a new flag to support this).

ref. https://github.com/kubernetes/kubernetes/issues/64742 and https://github.com/kubernetes/kubernetes/pull/64746

/cc @jpbetz @ptabor @xiang90 @jqmichael

---

@ptabor Seems like we've accidentally removed this code as we upgraded gRPC for etcd 3.5?

https://github.com/etcd-io/etcd/blob/4d9f1a96daa6be11383c1aa1e7840d9a078f6c0b/client/v3/doc.go#L102-L104

**Comment by ptabor** [comment]
@gyuho: 
 
It was not accidental. I think that the flag had nearly no control in v3.4. 

The flag was driving only global clientv3.lg logger: 

https://github.com/etcd-io/etcd/blob/2702f9e5f21193d8faffe3b6ae57dfc2dab662bd/clientv3/client.go#L50-L53

That was used only in 1 place. 

All other logging was done through class-level logger (`c.lg`) configured here: 

https://github.com/etcd-io/etcd/blob/2702f9e5f21193d8faffe3b6ae57dfc2dab662bd/clientv3/client.go#L430 


So I think that this flag should get replaced by generic mechanism of passing clientv3 config. 
I would consider a single flag that carries json and that json got merged with app-given spec using [JSON-merge-patch] (https://tools.ietf.org/html/rfc7386).

**Comment by gyuho** [comment]
> I think that the flag had nearly no control in v3.4.

It can be used to configure client-side logging level configurable (e.g., `zap.NewAtomicLevelAt(zap.DebugLevel)`)?

```
ETCD_CLIENT_DEBUG=1 ETCDCTL_API=3 etcdctl --endpoints ETCD_HOST:2379 get foo
```

will help us debug gRPC layer logging (e.g., `create subconn`).

Now, we cannot override this (unless we use `WithLogger` in Go)?

https://github.com/etcd-io/etcd/blob/63c51709b47043946bf2742a64338c7c6203e158/client/v3/client.go#L332-L340

We should add `ETCD_CLIENT_DEBUG` back with more generic solution.

**Comment by ptabor** [comment]
1. It does not require WithLogger.

The custom logging level can be configured using client-config & zap-config (that is JSON serializable): 
```
  lcfg := logutil.DefaultZapLoggerConfig
  lcfg.Level = zap.NewAtomicLevelAt(zap.DebugLevel)
  ccfg := clientv3.Config{LogConfig: &lcfg}
```

(I'm not saying that we shouldn't have env-veriable that drives the default behavior, just showing that this is not 
different from all the other properties configuring client).

2. As grpclog config is global (not per client/connection) and not-threadsafe,  we should reconfigure it on per-client basis. 
We might offer explicit helper methods how to configure it, but it should be explicit call from the user. 

I contributed some work-around for the thread-safety problem in 
https://github.com/grpc-ecosystem/go-grpc-middleware/pull/402, 
but its suitable for tests rather then implicit use in prod (where user can have other services and other preferred grpc config).

**Comment by ptabor** [comment]
Provided fix to maintain the feature in 3.5: https://github.com/etcd-io/etcd/pull/12786/commits
